File: src/com/bridgelabz/functional/SumofthreeInteger.java
/*
 * @purpose	:A program with cubic running time. Read in N integers and counts the
 * 			number of triples that sum to exactly 0
 * @author	:sangita awaghad
 * @since	:30-07-2019
 */
package com.bridgelabz.functional;

import com.bridgelabz.utility.Utility;

public class SumofthreeInteger {
	public static void main(String args[]) {
		int n;
		Utility utility=new Utility();
		System.out.println("How many Elemeny you want in array");
		n = utility.getInteger();

		int[] mArr = new int[n];
		System.out.println("Enter "+n+" Integer");
		for (int i = 0; i < n; i++) {
			mArr[i] = utility.getInteger();
		}
		findTriplet(mArr);
	}

	static void findTriplet(int[] arr) {
		int first;
		int second;
		int third ;
		for (first = 0; first < arr.length - 2; first++) {
			for (second = first + 1; first < arr.length - 1; first++) {
				for (third = second + 1; second < arr.length; second++) {
					if (arr[first] + arr[second] + arr[third] == 0) {
						System.out.println(arr[first] + " + " + arr[second] + " + " + arr[third] + " = 0");
					}
				}
			}
		}
	}
}

File: nodeProject/SumofThreeInteger.js
const utility=require('./Utility/Utility');

sumofThreeInteger=()=>{
    console.log(`how many element you want in array`);
    let size=utility.getInputNumber();
    let a=[];
    console.log(`enter the ${size} integer elements`);
    for(let i=0;i<size;i++)
    {
        a[i]=utility.getInputNumber();
    }
    for(let first=0;first<size-2;first++)
    {
        for(let second=first+1;second<size-1;second++)
        {
            for(let third=second+1;third<size;third++)
            {
                if(a[first]+a[second]+a[third]==0)
                {
                    console.log(`${a[first]}+${a[second]}+${a[third]}=0`);
                }
            }
        }
    }
}

module.exports=sumofThreeInteger();